python/modules/IcePy/Communicator.cpp

```diff
@@ -787,39 +790,52 @@ communicatorFlushBatchRequestsAsync(CommunicatorObject* self, PyObject* args, Py
 
     PyObjectHandle v = getAttr(compressBatch, "_value", false);
     assert(v.get());
-    Ice::CompressBatch cb = static_cast<Ice::CompressBatch>(PyLong_AsLong(v.get()));
+    Ice::CompressBatch compress = static_cast<Ice::CompressBatch>(PyLong_AsLong(v.get()));
 
     assert(self->communicator);
     const string op = "flushBatchRequests";
 
-    FlushAsyncCallbackPtr d = new FlushAsyncCallback(op);
-    Ice::Callback_Communicator_flushBatchRequestsPtr callback =
-        Ice::newCallback_Communicator_flushBatchRequests(d, &FlushAsyncCallback::exception, &FlushAsyncCallback::sent);
-
-    Ice::AsyncResultPtr result;
-
+    auto callback = make_shared<FlushAsyncCallback>(op);
+    function<void()> cancel;
     try
     {
-        result = (*self->communicator)->begin_flushBatchRequests(cb, callback);
+        cancel = (*self->communicator)->flushBatchRequestsAsync(
+            compress,
+            [callback](exception_ptr exptr)
+            {
+                try
+                {
+                    rethrow_exception(exptr);
+                }
+                catch (const Ice::Exception& ex)
+                {
+                    callback->exception(ex);
+                }
+                catch (...)
+                {
+                    assert(false);
+                }
```

Comment:
I'm not sure it's worth having this catch IMO, Ice shouldn't be throwing other exceptions, just let it crash.

Reply:
Removed